src/monster.cpp

```diff
@@ -1637,6 +1641,12 @@ void monster::process_effects()
         }
     }
 
+    if( has_effect("shrieking")){
+        sounds::sound(pos(), 120, _("a piercing wail!"), true);
+        // Lose 40% of our maximum speed while shrieking
+        set_speed_base( .6 * type->speed);
```

Comment:
Seconded, there are a lot of places that mess with monster speed that shouldn't.